src/main/java/com/znsio/rpi/properties/ReportPortalPropertiesOverloader.java

```diff
@@ -0,0 +1,169 @@
+        String[] getBranchNameCommand = new String[]{"git", "rev-parse", "--abbrev-ref", "HEAD"};
+        CommandLineResponse response = CommandLineExecutor.execCommand(getBranchNameCommand);
+        String branchName = response.getStdOut();
+        LOGGER.info(String.format("\tBranch name from git command: '%s': '%s'",
+                Arrays.toString(getBranchNameCommand), branchName));
+        return branchName;
+    }
+
+    private static String getBrowserVersion(String browserPath) {
+        String[] getBrowserVersionCommand = new String[]{browserPath, "-v", "|", "awk", "'{print $2}'"};
+        CommandLineResponse response = CommandLineExecutor.execCommand(getBrowserVersionCommand);
+        String BrowserVersion = response.getStdOut();
+        LOGGER.info(String.format("\tBrowser Version from CLI command: '%s': '%s'",
+                Arrays.toString(getBrowserVersionCommand), BrowserVersion));
+        return BrowserVersion;
+    }
+
+    private static boolean isPlatformWeb() {
+        return config.getProperty(Config.PLATFORM).equalsIgnoreCase("Web");
+    }
+
+    private static int getThreadCount() {
+        String xmlFilePath = System.getProperty("user.dir") + "/" + System.getProperty("suiteXmlFile");
+        Parser parser = new Parser(xmlFilePath);
+        XmlSuite xmlSuite;
+        try {
+            xmlSuite = parser.parseToList().get(0);
+        } catch (IOException ex) {
+            LOGGER.info("ERROR: Unable to find or load testng.xml file at location: "
+                    + xmlFilePath + "\n");
+            LOGGER.debug(ExceptionUtils.getStackTrace(ex));
+            throw new RuntimeException("ERROR: Unable to find or load testng.xml file at location: "
+                    + xmlFilePath + "\n", ex);
+        }
+        if ("none".equalsIgnoreCase(xmlSuite.getParallel().toString())) {
+            return DEFAULT_THREAD_COUNT;
+        }
+        return xmlSuite.getThreadCount();
+    }
+
+    private static void setRPAttributesFromSystemVariables() {
+        setRPAttributesFromEnvVariables();
+        setRPAttributesFromSystemProperties();
+    }
+
+    private static void setRPAttributesFromSystemProperties() {
+        Properties properties = System.getProperties();
+        Set<String> set = properties.stringPropertyNames();
+        for (String propkey : set) {
+            if (propkey.startsWith(RP_PREFIX)) {
+                addAttributes(getKeyWithoutPrefix(RP_PREFIX, propkey), properties.getProperty(propkey));
+            }
+        }
+    }
+
+    private static void setRPAttributesFromEnvVariables() {
+        Map<String, String> map = System.getenv();
+        for (String envKey : map.keySet()) {
+            if (envKey.startsWith(RP_PREFIX)) {
+                addAttributes(getKeyWithoutPrefix(RP_PREFIX, envKey), map.get(envKey));
+            }
+        }
+    }
+
+    private static String getKeyWithoutPrefix(String prefix, String key) {
```

Comment:
prefix is always RP_PREFIX. Not required to send as a parameter

Reply:
Updated the method parameters